core/src/ledger_cleanup_service.rs

```diff
@@ -223,6 +235,39 @@ impl LedgerCleanupService {
         Ok(())
     }
 
+    pub fn compact_ledger(
+        blockstore: &Arc<Blockstore>,
+        last_compaction_slot: &mut u64,
+        compaction_interval: u64,
+        highest_compact_slot: &Arc<AtomicU64>,
+        compaction_jitter: &mut u64,
+        max_jitter: Option<u64>,
+    ) {
+        let highest_compaction_slot = highest_compact_slot.load(Ordering::Relaxed);
+        if highest_compaction_slot.saturating_sub(*last_compaction_slot)
+            > (compaction_interval + *compaction_jitter)
+        {
+            info!(
+                "compacting data from slots {} to {}",
+                *last_compaction_slot, highest_compaction_slot,
+            );
+            if let Err(err) =
+                blockstore.compact_storage(*last_compaction_slot, highest_compaction_slot)
+            {
+                // This error is not fatal and indicates an internal error?
+                error!(
+                    "Error: {:?}; Couldn't compact storage from {:?} to {:?}",
+                    err, last_compaction_slot, highest_compaction_slot,
+                );
+            }
+            *last_compaction_slot = highest_compaction_slot;
+            let jitter = max_jitter.unwrap_or(0);
+            if jitter > 0 {
+                *compaction_jitter = thread_rng().gen_range(0, jitter);
```

Comment:
WDYT about also adding a `--jitter-step` which is used like `jitter = jitter_step * rand(0, max_jitter / jitter_step)`?  Idea being to improve the odds that compactions don't overlap in scenario where multiple nodes are serving a high-availability load balancer

Reply:
good idea, I'll add that

Reply:
we can do it in a follow-up if required